ParseJson.java

```diff
@@ -12,10 +12,65 @@
       messageParams.put(METHOD, jsonObj.getString(METHOD));
     } 
     if(jsonObj.has(RESULT)) {
-      messageParams.put(RESULT, jsonObj.get(RESULT));
+      JSONObject res = jsonObj.getJSONObject(RESULT);
```

Comment:
I'm not sure how to deal with this code in this shape if I were to work with it - it looks pretty tangled and hardly readable.
Have you considered using a library for JSON (Jackson or Gson?) to help alleviate the complexities of unmarshalling and keeping the code straightforward?
If not, can you sketch a test for this functionality?

Reply:
I second this, you really should not be doing any parsing of JSON yourself. If you need a schema, you can look into something [JSON Schema](https://json-schema.org) but that is also out of scope for this task.

Reply:
So I am not super familiar with JSON, but my understanding was that Jackson and Gson do not make JSON-RPC compliant JSON objects. Sasha and I discussed parsing in my first pull request here https://github.com/moyaog/PrototypeLinuxJobWorkerService/pull/1.

Reply:
I'm ok with this if this has been discussed and agreed upon.

Note though, JSON-RPC is **based** on JSON and there's nothing stopping you in producing or consuming objects with a JSON library in JSON-RPC format (which simply prescribes a specific field layout).

Reply:
Are there any other steps you would like me to take at at this point?

Reply:
You can merge this part as well.